internal/ingress/annotations/cors/main.go

```diff
@@ -123,16 +123,18 @@ func (c cors) Parse(ing *networking.Ingress) (interface{}, error) {
 
 	unparsedOrigins, err := parser.GetStringAnnotation("cors-allow-origin", ing)
 	if err == nil {
-		config.CorsAllowOrigin = strings.Split(unparsedOrigins, ",")
-		for i, origin := range config.CorsAllowOrigin {
+		origins := strings.Split(unparsedOrigins, ",")
+		for _, origin := range origins {
 			origin = strings.TrimSpace(origin)
 			if origin == "*" {
 				config.CorsAllowOrigin = []string{"*"}
 				break
 			}
-			if !corsOriginRegex.MatchString(origin) {
+
+			if corsOriginRegex.MatchString(origin) {
+				config.CorsAllowOrigin = append(config.CorsAllowOrigin, origin)
+			} else {
 				klog.Errorf("Error parsing cors-allow-origin parameters. Supplied incorrect origin: %s. Skipping.", origin)
-				config.CorsAllowOrigin = append(config.CorsAllowOrigin[:i], config.CorsAllowOrigin[i+1:]...)
 			}
 			klog.Infof("Current config.corsAllowOrigin %v", config.CorsAllowOrigin)
```

Comment:
this has nothing to do with the review (nice catch)
we could change this to the current origin instead of the whole list as well.

just a though

Reply:
I might be misinterpreting, but that was adjusting in this commit to use the current origin:
`config.CorsAllowOrigin = append(config.CorsAllowOrigin, origin)`, you might be looking at the old view?

Reply:
probably, i'm referring to this here: `klog.Infof("Current config.corsAllowOrigin %v", config.CorsAllowOrigin)` sorry about that :)